src/proxy/ITopic.ts

```diff
@@ -38,8 +38,15 @@ export interface ITopic<E> extends DistributedObject {
      *
      * More than one message listener can be added on one instance.
      *
+     * This method is for only Reliable Topic since this is a sync method.
+     * When this method is called from Topic, it throws {@link HazelcastError}.
+     * Check addListener method for using with Topic.
+     *
+     * `addMessageListener` is @deprecated since 5.3. Use {@link addListener} instead.
+     *
      * @param listener the MessageListener to add
      * @return registration ID
+     * @throws {HazelcastError} if it is used from Topic
      */
     addMessageListener(listener: MessageListener<E>): string;
```

Comment:
```suggestion
     * This method is for only Reliable Topic since this is a sync method.
     * When this method is called from Topic, it throws {@link HazelcastError}.
     * Check addListener method for using with Topic.
     *
     * @deprecated `addMessageListener` is deprecated since 5.3. Use {@link addListener} instead.
     *
     * @param listener the MessageListener to add
     * @return registration ID
     * @throws {@link HazelcastError} if it is used from Topic
     */
    addMessageListener(listener: MessageListener<E>): string;
```

Reply:
I suggest you to check the generated documentation as well.

Reply:
`{@link HazelcastError}` does not work like this. This is why I implemented it as `{HazelcastError}` . Others are done.

Reply:
No, it works.